lib/Optimizer/Transforms/QuakeSynthesizer.cpp

```diff
@@ -561,12 +632,21 @@ class QuakeSynthesizer
                   return builder.create<cudaq::cc::CastOp>(
                       loc, cudaq::cc::PointerType::get(stateTy), rawPtr);
                 });
-            continue;
+          } else if (stateData != nullptr && !stateData->isEmpty()) {
+            // Special case of running on a simulator in a different address
+            // space, when we have converted state data.
+            if (failed(synthesizeStateArgument(builder, module, counter,
+                                               argument, *stateData,
+                                               stateDataOffset)))
+              funcOp.emitOpError("synthesis failed for state*");
+            stateDataOffset++;
           } else {
-            funcOp.emitOpError("synthesis: unsupported argument type for "
-                               "remote devices and simulators: state*");
+            // All other cases are not yet supported (i.e. quantum hardware).
+            funcOp.emitOpError("synthesis: unsupported argument type on "
+                               "quantum devices: state*");
```

Comment:
Oh, it looks like you already updated the error message, 👍 . (Though I'm not sure how that other test was passing if the error message changed?)

Reply:
That test was failing, fixed